runtime/auth/aws-signing-common/common/src/aws/smithy/kotlin/runtime/auth/awssigning/AwsSigningConfig.kt

```diff
@@ -164,6 +164,14 @@ public class AwsSigningConfig(builder: Builder) {
      */
     public val expiresAfter: Duration? = builder.expiresAfter
 
+    /**
+     * Flag enabling whether detailed trace logging is enabled (if the signer implementation supports it). When true
+     * signers should emit intermediate logging details such as the canonical request and string to sign at the trace
+     * level. This is an opt-in configuration because these intermediate outputs may contain sensitive fields bound
+     * to headers, URI, or query parmaeters.
+     */
+    public val enableTraceLogging: Boolean = builder.enableTraceLogging
```

Comment:
Not sure what we wanted to call this, let the 🚲 🏚️  commence!

Reply:
I like this name *if* we were controlling all `logger.trace` calls with it, but we **always** log the "string to sign" at the trace level regardless of this flag. I think we should either rename this to better describe exactly what gets logged, or block the string-to-sign logging with this flag too.

Reply:
Yeah this is why I wanted to bikeshed over it because I had a similar feeling. I don't want to call it something too specific like `logCanonicalRequests` in the event there are other signing steps or calculations that we may want to put behind this flag.

Open to suggestions

Reply:
`enableSensitiveTraceLogging`?

Reply:
In tandem with my feedback above about `LogSigning` controlling _all_ signing logging, I think this should just be called `enableLogging`.